### PR TITLE
Add doc.go imports for staging/src/k8s.io/client-go/util/workqueue

### PR DESCRIPTION
**Pull request by Toper2035**
**What this PR does / why we need it**:

Adds canonical import paths to existing doc.go files for all packages under staging/src/k8s.io/sample-controller. This is so that we can use vanity URLs when importing staging packages.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

Only a partial fix for #68231. But breaking up PRs so they remain smaller.

**Special notes for your reviewer**:

I did not add canonical imports for the following files in client-go because they had a comment that said Code generated by client-gen. DO NOT EDIT.

**Release note**:

```release-note

NONE
```

### CODE REVIEW

**Comment by sttts** [comment]
I think these import comments are useful as they make the go compiler to fail if ppl import the packages in the wrong way.

For which packages do we want that? Why is the one modified here special?

**Comment by sttts** [comment]
@Toper2035 please squash the commits. Now there are 4 commits for one line that is changed.

**Comment by idealhack** [comment]
@Toper2035 And please sign the CLA as the bot said :)

**Comment by Toper2035** [comment]
updated the commit email, and it's ok now. @idealhack

**Comment by caesarxuchao** [comment]
/unassign

+1 to sttts's question https://github.com/kubernetes/kubernetes/pull/68272#issuecomment-418676025.

**Comment by idealhack** [comment]
@sttts As the description says this change belongs to #68231, for example, #68323 is waiting for review too.

File: staging/src/k8s.io/client-go/util/workqueue/doc.go [hunk 1]
```diff
@@ -23,4 +23,4 @@ limitations under the License.
 //  * Multiple consumers and producers. In particular, it is allowed for an
 //      item to be reenqueued while it is being processed.
 //  * Shutdown notifications.
-package workqueue
+package workqueue // import "k8s.io/client-go/util/workqueue"
```
Comment (nikhita) [inline staging/src/k8s.io/client-go/util/workqueue/doc.go]: @Toper2035 this is for client-go but the the title says that this for sample-controller.

https://github.com/kubernetes/kubernetes/pull/68783 updates the workqueue one. So can you fix the sample-controller ones in this? Thanks!

Comment (Toper2035) [reply staging/src/k8s.io/client-go/util/workqueue/doc.go]: @nikhita thanks, i have updated this PR's title.

**Pull request closed without merging**